sklearn/partial_dependence.py

```diff
@@ -0,0 +1,648 @@
+
+def partial_dependence(est, target_variables, grid=None, X=None, output=None,
+                       percentiles=(0.05, 0.95), grid_resolution=100,
+                       method=None):
+    """Partial dependence of ``target_variables``.
+
+    Partial dependence plots show the dependence between the joint values
+    of the ``target_variables`` and the function represented
+    by the ``est``.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model.
+    target_variables : array-like, dtype=int
+        The target features for which the partial dependency should be
+        computed (size should be smaller than 3 for visual renderings).
+    grid : array-like, shape=(n_points, len(target_variables))
+        The grid of ``target_variables`` values for which the
+        partial dependency should be evaluated (either ``grid`` or ``X``
+        must be specified).
+    X : array-like, shape=(n_samples, n_features)
+        The data on which ``est`` was trained. It is used to generate
+        a ``grid`` for the ``target_variables``. The ``grid`` comprises
+        ``grid_resolution`` equally spaced points between the two
+        ``percentiles``.
+    output : int, optional (default=None)
+        The output index to use for multi-output estimators.
+    percentiles : (low, high), default=(0.05, 0.95)
+        The lower and upper percentile used to create the extreme values
+        for the ``grid``. Only if ``X`` is not None.
+    grid_resolution : int, default=100
+        The number of equally spaced points on the ``grid``.
+    method : {'recursion', 'exact', 'estimated', None}, optional (default=None)
+        The method to use to calculate the partial dependence function:
+
+        - If 'recursion', the underlying trees of ``est`` will be recursed to
+          calculate the function. Only supported for BaseGradientBoosting and
+          ForestRegressor.
+        - If 'exact', the function will be calculated by calling the
+          ``predict_proba`` method of ``est`` for classification or ``predict``
+          for regression on ``X``for every point in the grid. To speed up this
+          method, you can use a subset of ``X`` or a more coarse grid.
+        - If 'estimated', the function will be calculated by calling the
+          ``predict_proba`` method of ``est`` for classification or ``predict``
+          for regression on the mean of ``X``.
+        - If None, then 'recursion' will be used if ``est`` is
+          BaseGradientBoosting or ForestRegressor, and 'exact' used for other
+          estimators.
+
+    Returns
+    -------
+    pdp : array, shape=(n_classes, n_points)
+        The partial dependence function evaluated on the ``grid``.
+        For regression and binary classification ``n_classes==1``.
+    axes : seq of ndarray or None
+        The axes with which the grid has been created or None if
```

Comment:
Thanks for your help! 

I am getting sklearn version `0.20.dev0` with no problem, however, the branch containing the latest changes on partial dependence plots [`f2e208e2f395f013bd54ed8dacbb8eb99db8ce09`](https://github.com/scikit-learn/scikit-learn/pull/5653/commits/f2e208e2f395f013bd54ed8dacbb8eb99db8ce09) is not coming through.

I used this command:

`pip3.4.exe install --user --upgrade git+https://github.com/scikit-learn/scikit-learn.git@f2e208e2f395f013bd54ed8dacbb8eb99db8ce09`

Which produces:
```
Collecting git+https://github.com/scikit-learn/scikit-learn.git@f2e208e2f395f013bd54ed8dacbb8eb99db8ce09
  Cloning https://github.com/scikit-learn/scikit-learn.git (to f2e208e2f395f013bd54ed8dacbb8eb99db8ce09) to c:\users\garcia~1\appdata\local\temp\pip-hr1yhagl-build
  Could not find a tag or branch 'f2e208e2f395f013bd54ed8dacbb8eb99db8ce09', assuming commit.
fatal: reference is not a tree: f2e208e2f395f013bd54ed8dacbb8eb99db8ce09
```
I also tried the user "trevorstephens", but there is no way I am getting the new `partial_dependence.py` file. I am sure I am missing something, but I do not know what. Please, any further suggestion? :-)




>